doc/source/ray-contribute/writing-code-snippets.rst

```diff
@@ -0,0 +1,189 @@
+.. _writing-code-snippets:
+
+==========================
+How to write code snippets
+==========================
+
+Docstrings and user guides should contain examples that illustrate Ray's APIs.
+
+This page describes how to write code snippets so that they're tested in CI.
+
+.. note::
+    The examples in this guide use reStructuredText. If you're writing
+    Markdown, use MyST syntax. To learn more, read the
+    `MyST documentation <https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#directives-a-block-level-extension-point>`_.
+
+-----------------
+Types of examples
+-----------------
+
+There are two types of examples: *doctest-style* and *code-output-style*.
+
+*doctest-style* examples
+========================
+
+*doctest-style* examples mimic interactive Python sessions. ::
+
+    .. doctest::
+
+        >>> def is_even(x):
+        ...     return (x % 2) == 0
+        >>> is_even(0)
+        True
+        >>> is_even(1)
+        False
+
+They're rendered like this:
+
+.. doctest::
+
+    >>> def is_even(x):
+    ...     return (x % 2) == 0
+    >>> is_even(0)
+    True
+    >>> is_even(1)
+    False
+
+.. tip::
+
+    If you're writing docstrings, exclude `.. doctest::` to simplify your code. ::
+
+        Example:
+            >>> def is_even(x):
+            ...     return (x % 2) == 0
+            >>> is_even(0)
+            True
+            >>> is_even(1)
+            False
+
+*code-output-style* examples
+============================
+
+*code-output-style* examples contain ordinary Python code. ::
+
+    .. testcode::
+
+        def is_even(x):
+                return (x % 2) == 0
+
+        print(is_even(0))
+        print(is_even(1))
+
+    .. testoutput::
+
+        True
+        False
+
+They're rendered like this:
+
+.. testcode::
+
+    def is_even(x):
+            return (x % 2) == 0
+
+    print(is_even(0))
+    print(is_even(1))
+
+.. testoutput::
+
+    True
+    False
+
+---------------------------------------
+Which type of example should you write?
+---------------------------------------
+
+If you're writing a small example that emphasizes object representations, use *doctest-style*. ::
```

Comment:
It might not be clear to everybody what an object representation is -- you could say something like "if you want to print intermediate objects"